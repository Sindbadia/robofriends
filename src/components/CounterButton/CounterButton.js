import React, { Fragment, PureComponent } from 'react'

class CounterButton extends PureComponent {
	state = {
		count: 0,
	}

	updateCount = () => {
		this.setState(prevState => ({ count: prevState.count + 1 }))
	}

	render() {
		return (
			<Fragment>
				<button id='counter' color={this.props.color} onClick={this.updateCount}>
					Count: {this.state.count}
				</button>
			</Fragment>
		)
	}
}

export default CounterButton
