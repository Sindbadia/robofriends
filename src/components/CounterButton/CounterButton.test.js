import { shallow } from 'enzyme'
import React from 'react'
import { create } from 'react-test-renderer'

import CounterButton from './CounterButton'

it('should render Card component', () => {
	const mockColor = 'red'

	expect(create(<CounterButton color={mockColor} />)).toMatchSnapshot()
})

it('should correctly increments the counter', () => {
	const mockColor = 'red'
	const wrapper = shallow(<CounterButton color={mockColor} />)

	wrapper.find('[id="counter"]').simulate('click')
	expect(wrapper.state()).toEqual({ count: 1 })
	expect(wrapper.props().children.props.color).toEqual('red')
})
