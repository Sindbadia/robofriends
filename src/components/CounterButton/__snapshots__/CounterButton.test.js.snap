// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Card component 1`] = `
<button
  color="red"
  id="counter"
  onClick={[Function]}
>
  Count: 
  0
</button>
`;
